olive/systems/docker/docker_system.py

```diff
@@ -171,5 +172,11 @@ def _run_container(self, tempdir, model: OliveModel, metrics: List[Metric], cont
                 )
                 logger.debug("Dev mount cleaned up successfully")
```

Comment:
@xiaoyu-work , why we need start another docker to clean up the mount path? for the mount path in docker container, the container is already removed. for the host path, it is temp folder which will be automatically removed?

Reply:
I didn't add this. Seems @jambayk added this for cleaning dev mount https://github.com/microsoft/Olive/pull/210. I have a same question here. Can we just remove the container?

Reply:
@jambayk

Reply:
@trajepl , it seems [dev_mount_cleanup.py](https://github.com/microsoft/Olive/pull/210/files#diff-b6ef65dfa0431903cc41c3a9a6ecc495b211f1844bb33a6c83e9919085c26abc) could be removed and so did create_dev_cleanup_mount

Reply:
This part from #210 is to restart the container with `root` user to remove the generated files in container with `root` permission

This image shows the host case which will create the new `pycache` with local users.
![image](https://github.com/microsoft/Olive/assets/13343117/c8cbaf68-f49f-4c92-a203-9f1e331e5f31)

But when it comes to container, Olive run container with `root` users who will generated the file in mounted path which cannot be removed when the context of temp directory ends.

Reply:
Run docker with `--user $(id -u):$(id -g)` would help. But there is another issue when the docker is started with host user but the user is not in sudo group, huggingface model download will fail as they will cache the model under `/.cache/xxx` which is the root path. 

I will start a new PR if there is another elegant way.

Reply:
I added the dev mount clean up because of the reason Jiapeng mentioned above. Even thought we mount a temporary directory, it cannot automatically clean up the pycache which is created with root permissions by the container.